src/5.0_to_5.1/calculated-measures/migrateCalculatedMeasures.ts

```diff
@@ -47,8 +51,14 @@ export function migrateCalculatedMeasures(
   } = legacyCalculatedMeasuresFolder?.children
     ? _mapKeys(
         legacyCalculatedMeasuresFolder.children.content.children,
-        (value, key) =>
-          getCalculatedMeasureName(legacyCalculatedMeasuresFolder, key),
+        (value, key) => {
+          const calculatedMeasureName = getCalculatedMeasureName(
+            legacyCalculatedMeasuresFolder,
+            key,
+          );
+          calculatedMeasureNamesAndIds[calculatedMeasureName] = key;
```

Comment:
The starting point is the id.
So I'd find it more logical to have a single map where all the information is stored:
```
const namesOfCalculatedMeasuresToMigrate: string[] = [];
 const legacyCalculatedMeasures: {
    [id: string]: {record: ContentRecord, name: string};
  } = {}
_forEach(legacyCalculatedMeasuresFolder.children.content.children, (record, id) => {
 const name = getCalculatedMeasureName(legacyCalculatedMeasuresFolder, id);
  namesOfCalculatedMeasuresToMigrate.push(name);
  legacyCalculatedMeasures[id] = {name, record};
})
```